DigitalEcosystem/publication/exfoliation2DMaterials/2dm_exfoliation.py

```diff
@@ -0,0 +1,567 @@
+
+# In[]:
+
+
+roost_train_results = pd.read_csv("roost/roost_train_predictions.csv", index_col="material_id")
+roost_test_results  = pd.read_csv("roost/roost_test_predictions.csv", index_col="material_id")
+
+
+# In[]:
+
+
+DigitalEcosystem.utils.figures.publication_parity_plot(train_y_true = roost_train_results['exfoliation_energy_target'],
+                                                       train_y_pred =  roost_train_results['exfoliation_energy_pred_n0'],
+                                                       test_y_true = roost_test_results['exfoliation_energy_target'],
+                                                       test_y_pred = roost_test_results['exfoliation_energy_pred_n0'],
+                                                       axis_label = "Exfoliation Energy (J/m^2)",
+                                                       filename = "roost_2dm_exfoliation_parity.jpeg")
+
+
+# In[]:
+
+
+print("Test Set Error Metrics")
+for key, fun in metrics.items():
+    value = fun(y_true=roost_test_results['exfoliation_energy_target'], y_pred=roost_test_results['exfoliation_energy_pred_n0'])
+    print(key,np.round(value,4))
+    
+print("\nTraining Set Error Metrics")
+for key, fun in metrics.items():
+    value = fun(y_true=roost_train_results['exfoliation_energy_target'], y_pred=roost_train_results['exfoliation_energy_pred_n0'])
+    print(key,np.round(value,4))
+
+
+# # SISSO
+# 
+# SISSO is a symbolic regression technique focused on creating interpretable machine learning models. 
+# 
+# Due to the exponential computational cost of running a SISSO model as the number of features and rungs increases, we need to restrict the feature space. To do that, we'll use LASSO-based feature selection (essentially we can look at how quickly LASSO extinguishes a variable to get an idea of its importance). 
+
+# In[]:
+
+
+sisso_feature_selector = sklearn.feature_selection.SelectFromModel(sklearn.linear_model.LassoCV(random_state=RANDOM_SEED),
+                                                                   threshold=-np.inf,
+                                                                   max_features=16,
+                                                                   prefit=False)
+sisso_feature_selector.fit(train_x, train_y.ravel())
+
+sisso_features = [col for (col, is_selected) in zip(train[descriptors].columns, sisso_feature_selector.get_support()) if is_selected]
+print("\n".join(sisso_features))
+
+
+# In[]:
+
+
+sisso_dir = "./sisso"
+os.makedirs(sisso_dir, exist_ok=True)
+
+sisso_data_train = train[target_column + sisso_features]
+sisso_data_test = test[target_column + sisso_features]
+
+sisso_data_train.to_csv(os.path.join(sisso_dir, 'sisso_train.csv'), index_label='2dm_id (unitless)')
+
+
+# At this point, a SISSO model was run.
+
+# In[]:
+
+
+sisso_models = {
+    'r1_1term': lambda df: 1.159576043776266e+00 + \
```

Comment:
Same as https://github.com/Exabyte-io/Scientific-Projects/pull/29/files?authenticity_token=NsQC7vJVfK9clkiaFdPHBomzqKJ0AkYZteHpASCHq5d%2FAsQYsoU2nsc53%2FT012N6kxqU4bZElfbOVaGu7G%2FlDw%3D%3D&file-filters%5B%5D=.py#r724596674 - let's spell out where the numbers came from

Reply:
Added some commentary above the cell talking about where they're from

Reply:
Let's explain that the `r1_2term` are extracted from `rung1_log.txt` for clarity, and `(ave:atomic_volume - ave:Polarizability)` is what SISSO outputs.

Also - `(ave:atomic_volume - ave:Polarizability)` doesn't make sense from the units perspective - why is that?

Reply:
Per our discussion - let's add `.dat` files and explain how we extract the coefficeints

Reply:
Putting the .dat files in from the 5066 branch, since I'm re-running these anyway

Reply:
Oh also, turns out that the units for that comparison do make sense. Based on values of a couple elemental system, looks like they're using the polarizability constant to indicate polarizability. That's got units of length^3. NIST has a few values here - https://cccbdb.nist.gov/pollistx.asp